src/t_set.c

```diff
@@ -995,16 +1009,40 @@ void sinterGenericCommand(client *c, robj **setkeys,
 
 /* SINTER key [key ...] */
 void sinterCommand(client *c) {
-    sinterGenericCommand(c,c->argv+1,c->argc-1,NULL,0);
+    sinterGenericCommand(c, c->argv+1,  c->argc-1, NULL, 0, 0);
 }
 
+/* SINTERCARD numkeys key [key ...] [LIMIT limit] */
 void sinterCardCommand(client *c) {
-    sinterGenericCommand(c,c->argv+1,c->argc-1,NULL,1);
+    long j;
+    long numkeys = 0; /* Number of keys. */
+    long limit = 0;   /* 0 means not limit. */
+
+    if (getRangeLongFromObjectOrReply(c, c->argv[1], 1, LONG_MAX,
+                                      &numkeys, "numkeys should be greater than 0") != C_OK)
+        return;
+
+    for (j = 2 + numkeys; j < c->argc; j++) {
+        char *opt = c->argv[j]->ptr;
+        int moreargs = (c->argc - 1) - j;
+
+        if (!strcasecmp(opt, "LIMIT") && moreargs) {
+            j++;
+            if (getRangeLongFromObjectOrReply(c, c->argv[j], 1, LONG_MAX,
+                                              &limit, "limit should be greater than 0") != C_OK)
+                return;
```

Comment:
any reason not to allow setting LIMIT explicitly to 0? maybe it can make it easier for client libraries to wrap this command.
if we do that, we can use `getPositiveLongFromObjectOrReply`

Reply:
I originally thought that if you want to use LIMIT, you mast pass the value.
But there is no harm passing 0. I changed it